sdcm/nemesis.py

```diff
@@ -1876,6 +1881,43 @@ def disrupt_grow_shrink_cluster(self):
 
         self.log.info("Finish cluster shrink. Current number of nodes %s", len(self.cluster.nodes))
 
+    def disrupt_hot_reloading_internode_certificate(self):
+        '''
+        https://github.com/scylladb/scylla/issues/6067
+        Scylla has the ability to hot reload SSL certificates.
+        This test will create and reload new certificates for the inter node communication.
+        '''
+        self._set_current_disruption('HotReloadInternodeCertificate')
+
+        @retrying(allowed_exceptions=LogContentNotFound)
+        def check_ssl_reload_log(target_node, file_path):
+            msg = target_node.search_database_log(f"Reload.*{file_path}",
+                                                  start_from_beginning=False, severity=Severity.NORMAL)
+            if not msg:
+                raise LogContentNotFound('Reload SSL message not found in node log')
+            return msg
+
+        if not self.cluster.params.get('server_encrypt', None):
+            raise UnsupportedNemesis('Server Encryption is not enabled, hence skipping')
+
+        ssl_files_location = '/etc/scylla/ssl_conf'
+        in_place_crt = self.target_node.remoter.run(f"cat {os.path.join(ssl_files_location, 'db.crt')}",
+                                                    ignore_status=True).stdout
+
+        update_certificates()
```

Comment:
I guess that every time you get a new certificate and on a second run of the nemesis it will use a new certificate, is that correct?

Reply:
always it is called, it create a new certificate, but it runs locally, so next steps in the nemesis is to copy it to all nodes, and then, place the file in the folder Scylla is listening.